src/bos/common/clients/cfs/components.py

```diff
@@ -0,0 +1,113 @@
+#
+# MIT License
+#
+# (C) Copyright 2021-2022, 2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+from collections import defaultdict
+import logging
+
+from .base import BasePagedCfsEndpoint
+
+LOGGER = logging.getLogger(__name__)
+
+GET_BATCH_SIZE = 200
+PATCH_BATCH_SIZE = 1000
+
+
+class ComponentEndpoint(BasePagedCfsEndpoint):
+    ENDPOINT = 'components'
+    ITEM_FIELD_NAME = 'components'
+
+    def get_components(self, **kwargs):
+        return self.get_items(**kwargs)
+
+    def patch_components(self, data):
+        return self.update_items(data)
+
+    def get_components_from_id_list(self, id_list):
+        if not id_list:
+            LOGGER.warning(
+                "get_components_from_id_list called without IDs; returning without action."
```

Comment:
Function name should not be part of the logs.
Log can be 
"No IDs were provided, valid list of IDs are required."

Reply:
This is part of the existing function that was copied to the new location. As we discussed yesterday, I don't plan to use this PR to make those kind of changes to the existing functions. We can handle them in separate PRs if warranted.